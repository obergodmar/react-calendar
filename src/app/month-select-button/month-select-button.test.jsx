import React from "react";
import { shallow } from "enzyme";
import { describe, jest, test } from "@jest/globals";

import { MonthSelectButton } from ".";

describe("Month Select Button", () => {
  const handleClick = jest.fn();
  const currentDate = "January 1970";
  const handleAddMonth = jest.fn();
  const wrapper = shallow(
    <MonthSelectButton
      onClick={handleClick}
      currentDate={currentDate}
      addMonth={handleAddMonth}
      isShown={false}
    />
  );

  test("Should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should remove month by click", () => {
    wrapper.find(".buttonUp").simulate("click");
    expect(handleAddMonth).toHaveBeenCalledWith(-1);
  });

  test("Should add month by click", () => {
    wrapper.find(".buttonDown").simulate("click");
    expect(handleAddMonth).toHaveBeenCalledWith(1);
  });
});
