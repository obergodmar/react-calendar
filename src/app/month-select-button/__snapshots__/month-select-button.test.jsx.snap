// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month Select Button Should match snapshot 1`] = `
<div
  className="buttonGroup"
>
  <button
    className="monthSelectButton"
    onClick={[MockFunction]}
  >
    <span>
      January 1970
    </span>
    <span
      className="selectIcon"
    />
  </button>
  <div
    className="buttonUpDown"
  >
    <button
      className="buttonUp"
      onClick={[Function]}
    />
    <button
      className="buttonDown"
      onClick={[Function]}
    />
  </div>
</div>
`;
